libs/dao-data/src/utils/proposals.ts

```diff
@@ -0,0 +1,53 @@
+import { PROPOSAL_STATUSES } from '@daohaus/common-utilities';
+import { Proposal } from '../types';
+
+export const getProposalStatus = (proposal: Partial<Proposal>): string => {
+  if (!proposal.sponsored) {
+    return PROPOSAL_STATUSES.unsponsored;
+  }
+  if (proposal.cancelled) {
+    return PROPOSAL_STATUSES.cancelled;
+  }
+  if (proposal.passed) {
+    return PROPOSAL_STATUSES.passed;
+  }
+  if (proposal.actionFailed) {
+    return PROPOSAL_STATUSES.actionFailed;
+  }
+
+  const now = new Date().getTime() / 1000;
+  if (
+    Number(proposal.votingStarts) < now &&
```

Comment:
Refactoring all these 'if' statements to functions (ex. if(isVotingPeriod(proposal)) ) will: 

- Make this more readable
-  Allow us to reuse this functionality later on in the UI
- Ensure that the methods we use to get proposal status are consistent across all of our apps.

Reply:
I like that!